### ISSUE TITLE
Python 3.10 compatibility error

### ISSUE THREAD

**Issue by ondrejj** [nandoflorestan/kajiki#49]
Unable to use kajiki on python3.10b2 (b1 works, b2 don't).
Here is an example:
```
import kajiki

tpl = kajiki.TextTemplate('''%def quote(caller, speaker)
    %for i in range(2)
Quoth $speaker, "${caller(i)}."
    %end
%end
%call(n) quote(%caller ,'the raven')
Nevermore $n\\
%end''')
```

Raises an exception:

```
Traceback (most recent call last):
  File "/root/Kajiki-0.8.2/kajiki/template.py", line 323, in from_ir
    exec(py_text, dct)
  File "<string>", line 17
    yield caller(i)}.
                   ^
SyntaxError: unmatched '}'
```

Looks like problem is in kajiki/text.py:
`compile(self.source[self.pos:], '', 'eval')`,
and python3.10 better error handling. I don't know, how to fix this, compile raises error from end of string instead of from start.

Short python example:
```
Python 3.10.0b1 (default, May  4 2021, 00:00:00) [GCC 11.1.1 20210428 (Red Hat 11.1.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> compile('caller(i)}."', '', 'eval')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "", line 1
    caller(i)}."
             ^
SyntaxError: unmatched '}'
```

```
Python 3.10.0b2 (default, Jun  1 2021, 00:00:00) [GCC 11.1.1 20210531 (Red Hat 11.1.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> compile('caller(i)}."', '', 'eval')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "", line 1
    caller(i)}."
               ^
SyntaxError: unterminated string literal (detected at line 1)
>>>
```

Any volunteer, who can fix this? Fedora's kajiki build is broken. :-(

**Comment by hroncok** [comment]
```python
compile('caller(i)}."', '', 'eval')
```

This will always fail, even on Python 3.9. I don't think the new exception text is the problem here.
The problem seem to be that the chunk is split at a weird location.

**Comment by ondrejj** [comment]
Always fails, but with different message. Here is the code of kajiki/text.py:
```
    def _get_braced_expr(self):
        try:
            compile(self.source[self.pos:], '', 'eval')
        except SyntaxError as se:
            end = self.pos + sum([se.offset] + [len(line) + 1
                                                for idx, line in enumerate(self.
                                                if idx < se.lineno - 1])
            text = self.source[self.pos:end - 1]
            self.pos = end
            return self.expr(text)
```
When it fails in python3.10, exception's "se.offset" is different. Python3.10's error message is from end of this string and end position is computed differently, what causes to fail in kajiki tests.

**Comment by ondrejj** [comment]
A short testing script, which displays, what's different:

```
import sys
print(sys.version)
try:
  compile('caller(i)}."', '', 'eval')
except SyntaxError as se:
  print(se)
  print(se.offset)
```

differences:

```
3.10.0b1 (default, May  4 2021, 00:00:00) [GCC 11.1.1 20210428 (Red Hat 11.1.1-1)]
unmatched '}' (, line 1)
10
```

```
3.10.0b2 (default, Jun  1 2021, 00:00:00) [GCC 11.1.1 20210531 (Red Hat 11.1.1-3)]
unterminated string literal (detected at line 1) (, line 1)
12
```

**Comment by jackrosenthal** [comment]
Fixed by #52
